From 4bfd09cc6b3517483bbb0033dd2168e28bcad3f1 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Fri, 24 Jun 2016 17:32:57 +0200
Subject: [PATCH 527/538] ARM: mvebu: add support for Armada-395-GP-GW and
 Armada-395-GP-EAP boards

It occurs that there are 2 GP boards:
'armada-395-gp-eap' and 'armada-395-gp-gw'
This commit introduces two new dts for each baord and abstract common
nodes to: armada-395-gp.dtsi

Change-Id: I53a3a35927205c4e220fb4d949eae8173f874020
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30721
Reviewed-by: Lior Amsalem <alior@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
Verified-Armada38x-39x: Lior Amsalem <alior@marvell.com>
---
 .../devicetree/bindings/arm/marvell/armada-39x.txt |   3 +-
 arch/arm/boot/dts/armada-395-gp-eap.dts            |  65 ++++++
 arch/arm/boot/dts/armada-395-gp-gw.dts             |  65 ++++++
 arch/arm/boot/dts/armada-395-gp.dts                | 236 ---------------------
 arch/arm/boot/dts/armada-395-gp.dtsi               | 220 +++++++++++++++++++
 5 files changed, 352 insertions(+), 237 deletions(-)
 create mode 100644 arch/arm/boot/dts/armada-395-gp-eap.dts
 create mode 100644 arch/arm/boot/dts/armada-395-gp-gw.dts
 delete mode 100644 arch/arm/boot/dts/armada-395-gp.dts
 create mode 100644 arch/arm/boot/dts/armada-395-gp.dtsi

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-39x.txt b/Documentation/devicetree/bindings/arm/marvell/armada-39x.txt
index 48bf9a3..832ff35 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-39x.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-39x.txt
@@ -18,4 +18,5 @@ compatible: must contain "marvell,armada398"
 Example:
 
 compatible = "marvell,a398-db", "marvell,a390-db", "marvell,a395-amc",
-	     "marvell,a395-gp", "marvell,armada398", "marvell,armada395", "marvell,armada390";
+	     "marvell,a395-gp-gw", "marvell,a395-gp-eap", "marvell,armada398",
+	     "marvell,armada395", "marvell,armada390";
diff --git a/arch/arm/boot/dts/armada-395-gp-eap.dts b/arch/arm/boot/dts/armada-395-gp-eap.dts
new file mode 100644
index 0000000..b0502f0
--- /dev/null
+++ b/arch/arm/boot/dts/armada-395-gp-eap.dts
@@ -0,0 +1,65 @@
+/*
+ * Device Tree file for Marvell Armada 395 GP EAP board
+ * (DB-88F6925-EAP-10G)
+ *
+ * Copyright (C) 2016 Marvell
+ *
+ * Grzegorz Jaszczyk <jaz@semihalf.com>
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is licensed under the terms of the GNU General Public
+ *     License version 2.  This program is licensed "as is" without
+ *     any warranty of any kind, whether express or implied.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/dts-v1/;
+#include "armada-395-gp.dtsi"
+
+/ {
+	model = "Marvell Armada 395 GP EAP Board";
+	compatible = "marvell,a395-gp-eap", "marvell,armada395", "marvell,armada380";
+
+	soc {
+		gop {
+			emac3: mac3 {
+				phy = <&phy3>;
+				phy-mode = "sgmii";
+				force-link = "yes";
+			};
+		};
+
+		pp3_platform {
+			nic@3 {
+				status = "okay";
+				emac-data = <&emac3>;
+			};
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/armada-395-gp-gw.dts b/arch/arm/boot/dts/armada-395-gp-gw.dts
new file mode 100644
index 0000000..b5f471a
--- /dev/null
+++ b/arch/arm/boot/dts/armada-395-gp-gw.dts
@@ -0,0 +1,65 @@
+/*
+ * Device Tree file for Marvell Armada 395 GP GW board
+ * (DB-88F6925-GW-2.5G)
+ *
+ * Copyright (C) 2016 Marvell
+ *
+ * Grzegorz Jaszczyk <jaz@semihalf.com>
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is licensed under the terms of the GNU General Public
+ *     License version 2.  This program is licensed "as is" without
+ *     any warranty of any kind, whether express or implied.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/dts-v1/;
+#include "armada-395-gp.dtsi"
+
+/ {
+	model = "Marvell Armada 395 GP GW Board";
+	compatible = "marvell,a395-gp-gw", "marvell,armada395", "marvell,armada380";
+
+	soc {
+		gop {
+			emac2: mac2 {
+				phy = <&phy2>;
+				phy-mode = "sgmii";
+				force-link = "yes";
+			};
+		};
+
+		pp3_platform {
+			nic@2 {
+				status = "okay";
+				emac-data = <&emac2>;
+			};
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/armada-395-gp.dts b/arch/arm/boot/dts/armada-395-gp.dts
deleted file mode 100644
index 9bc6082..0000000
--- a/arch/arm/boot/dts/armada-395-gp.dts
+++ /dev/null
@@ -1,236 +0,0 @@
-/*
- * Device Tree file for Marvell Armada 395 GP board
- * (DB-88F6925-GW-2.5G)
- *
- * Copyright (C) 2014 Marvell
- *
- * Grzegorz Jaszczyk <jaz@semihalf.com>
- *
- * This file is dual-licensed: you can use it either under the terms
- * of the GPL or the X11 license, at your option. Note that this dual
- * licensing only applies to this file, and not this project as a
- * whole.
- *
- *  a) This file is licensed under the terms of the GNU General Public
- *     License version 2.  This program is licensed "as is" without
- *     any warranty of any kind, whether express or implied.
- *
- * Or, alternatively,
- *
- *  b) Permission is hereby granted, free of charge, to any person
- *     obtaining a copy of this software and associated documentation
- *     files (the "Software"), to deal in the Software without
- *     restriction, including without limitation the rights to use,
- *     copy, modify, merge, publish, distribute, sublicense, and/or
- *     sell copies of the Software, and to permit persons to whom the
- *     Software is furnished to do so, subject to the following
- *     conditions:
- *
- *     The above copyright notice and this permission notice shall be
- *     included in all copies or substantial portions of the Software.
- *
- *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
- *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
- *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
- *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
- *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- *     OTHER DEALINGS IN THE SOFTWARE.
- */
-
-/dts-v1/;
-#include "armada-395.dtsi"
-
-/ {
-	model = "Marvell Armada 395 GP Board";
-	compatible = "marvell,a395-gp", "marvell,armada395", "marvell,armada380";
-
-	chosen {
-		bootargs = "console=ttyS0,115200 earlyprintk";
-	};
-
-	memory {
-		device_type = "memory";
-		reg = <0x00000000 0x10000000>; /* 256 MB */
-	};
-
-	soc {
-		ranges = <MBUS_ID(0xf0, 0x01) 0 0xf1000000 0x100000
-			  MBUS_ID(0x01, 0x1d) 0 0xfff00000 0x100000
-				/* nss window, 16M */
-			  MBUS_ID(NSS_TARGET_ID, NSS_REGS_ATTR)  0 0xf5000000 NSS_REGS_SIZE
-				/* mac_nic window, (nss internal window) 128M */
-			  MBUS_ID(NSS_TARGET_ID, NSS_SPACE_ATTR) 0 0xD0000000 NSS_SPACE_SIZE>;
-
-		internal-regs {
-			spi0: spi@10600 {
-				status = "disabled";
-				pinctrl-0 = <&spi0_pins>;
-				pinctrl-names = "default";
-			};
-
-			spi@10680 {
-				status = "disabled";
-
-				spi-flash@1 {
-					#address-cells = <1>;
-					#size-cells = <1>;
-					compatible = "w25q128", "jedec,spi-nor";
-					reg = <0>; /* Chip select 0 */
-					spi-max-frequency = <108000000>;
-
-					partition@0 {
-						label = "U-Boot";
-						reg = <0 0x400000>;
-					};
-					partition@400000 {
-						label = "Filesystem";
-						reg = <0x400000 0xc00000>;
-					};
-				};
-			};
-
-			i2c@11000 {
-				status = "okay";
-				clock-frequency = <100000>;
-			};
-
-			mdio {
-				phy1: ethernet-phy@1 {
-					reg = <5>;
-				};
-
-				phy2: ethernet-phy@2 {
-					reg = <4>;
-				};
-
-				phy3: ethernet-phy@3 {
-					reg = <16>;
-				};
-			};
-
-			pinctrl@18000 {
-				spi0_pins: spi0_pins {
-					marvell,pins = "mpp15", "mpp16", "mpp17",
-						       "mpp18";
-					marvell,function = "spi0";
-				};
-			};
-
-			serial@12000 {
-				status = "okay";
-			};
-
-			usb@58000 {
-				status = "okay";
-			};
-
-			sata@a8000 {
-				status = "okay";
-			};
-
-			flash@d0000 {
-				status = "okay";
-				pinctrl-0 = <&nand_pins>;
-				pinctrl-names = "default";
-				num-cs = <1>;
-				marvell,nand-keep-config;
-				marvell,nand-enable-arbiter;
-				nand-on-flash-bbt;
-				nand-ecc-strength = <4>;
-				nand-ecc-step-size = <512>;
-
-				partition@0 {
-					label = "U-Boot";
-					reg = <0x00000000 0x00600000>;
-					read-only;
-				};
-
-				partition@800000 {
-					label = "uImage";
-					reg = <0x00600000 0x00400000>;
-					read-only;
-				};
-
-				partition@1000000 {
-					label = "Root";
-					reg = <0x00a00000 0x3f600000>;
-				};
-			};
-
-			sdhci@d8000 {
-				clock-frequency = <200000000>;
-				broken-cd;
-				wp-inverted;
-				bus-width = <8>;
-				status = "okay";
-				no-1-8-v;
-			};
-
-			usb3@f0000 {
-				status = "okay";
-			};
-		};
-
-		nss_complex {
-			status = "okay";
-			pinctrl-0 = <&smi_nss_pins>;
-			pinctrl-names="default";
-		};
-
-		gop {
-			emac0: mac0 {
-				phy-mode = "rxaui";
-			};
-
-			emac3: mac3 {
-				phy = <&phy3>;
-				phy-mode = "sgmii";
-				force-link = "yes";
-			};
-		};
-
-		pp3_platform {
-			status = "okay";
-
-			nic@0 {
-				status = "okay";
-				emac-data = <&emac0>;
-			};
-
-			nic@3 {
-				status = "okay";
-				emac-data = <&emac3>;
-			};
-
-			nss@0 {
-				status = "okay";
-			};
-		};
-
-		pcie-controller {
-			status = "okay";
-			/*
-			 * The three PCIe units are accessible through
-			 * standard PCIe slots on the board.
-			 */
-			pcie@1,0 {
-				/* Port 0, Lane 0 */
-				status = "disabled";
-			};
-			pcie@2,0 {
-				/* Port 1, Lane 0 */
-				status = "okay";
-			};
-			pcie@3,0 {
-				/* Port 2, Lane 0 */
-				status = "disabled";
-			};
-			pcie@4,0 {
-				/* Port 3, Lane 0 */
-				status = "okay";
-			};
-		};
-	};
-};
diff --git a/arch/arm/boot/dts/armada-395-gp.dtsi b/arch/arm/boot/dts/armada-395-gp.dtsi
new file mode 100644
index 0000000..668eed0
--- /dev/null
+++ b/arch/arm/boot/dts/armada-395-gp.dtsi
@@ -0,0 +1,220 @@
+/*
+ * Device Tree Include file for Marvell Armada 395 GP boards
+ *
+ * Copyright (C) 2014 Marvell
+ *
+ * Grzegorz Jaszczyk <jaz@semihalf.com>
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is licensed under the terms of the GNU General Public
+ *     License version 2.  This program is licensed "as is" without
+ *     any warranty of any kind, whether express or implied.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "armada-395.dtsi"
+
+/ {
+	chosen {
+		bootargs = "console=ttyS0,115200 earlyprintk";
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x00000000 0x10000000>; /* 256 MB */
+	};
+
+	soc {
+		ranges = <MBUS_ID(0xf0, 0x01) 0 0xf1000000 0x100000
+			  MBUS_ID(0x01, 0x1d) 0 0xfff00000 0x100000
+				/* nss window, 16M */
+			  MBUS_ID(NSS_TARGET_ID, NSS_REGS_ATTR)  0 0xf5000000 NSS_REGS_SIZE
+				/* mac_nic window, (nss internal window) 128M */
+			  MBUS_ID(NSS_TARGET_ID, NSS_SPACE_ATTR) 0 0xD0000000 NSS_SPACE_SIZE>;
+
+		internal-regs {
+			spi0: spi@10600 {
+				status = "disabled";
+				pinctrl-0 = <&spi0_pins>;
+				pinctrl-names = "default";
+			};
+
+			spi@10680 {
+				status = "disabled";
+
+				spi-flash@1 {
+					#address-cells = <1>;
+					#size-cells = <1>;
+					compatible = "w25q128", "jedec,spi-nor";
+					reg = <0>; /* Chip select 0 */
+					spi-max-frequency = <108000000>;
+
+					partition@0 {
+						label = "U-Boot";
+						reg = <0 0x400000>;
+					};
+					partition@400000 {
+						label = "Filesystem";
+						reg = <0x400000 0xc00000>;
+					};
+				};
+			};
+
+			i2c@11000 {
+				status = "okay";
+				clock-frequency = <100000>;
+			};
+
+			mdio {
+				phy1: ethernet-phy@1 {
+					reg = <5>;
+				};
+
+				phy2: ethernet-phy@2 {
+					reg = <4>;
+				};
+
+				phy3: ethernet-phy@3 {
+					reg = <16>;
+				};
+			};
+
+			pinctrl@18000 {
+				spi0_pins: spi0_pins {
+					marvell,pins = "mpp15", "mpp16", "mpp17",
+						       "mpp18";
+					marvell,function = "spi0";
+				};
+			};
+
+			serial@12000 {
+				status = "okay";
+			};
+
+			usb@58000 {
+				status = "okay";
+			};
+
+			sata@a8000 {
+				status = "okay";
+			};
+
+			flash@d0000 {
+				status = "okay";
+				pinctrl-0 = <&nand_pins>;
+				pinctrl-names = "default";
+				num-cs = <1>;
+				marvell,nand-keep-config;
+				marvell,nand-enable-arbiter;
+				nand-on-flash-bbt;
+				nand-ecc-strength = <4>;
+				nand-ecc-step-size = <512>;
+
+				partition@0 {
+					label = "U-Boot";
+					reg = <0x00000000 0x00600000>;
+					read-only;
+				};
+
+				partition@800000 {
+					label = "uImage";
+					reg = <0x00600000 0x00400000>;
+					read-only;
+				};
+
+				partition@1000000 {
+					label = "Root";
+					reg = <0x00a00000 0x3f600000>;
+				};
+			};
+
+			sdhci@d8000 {
+				clock-frequency = <200000000>;
+				broken-cd;
+				wp-inverted;
+				bus-width = <8>;
+				status = "okay";
+				no-1-8-v;
+			};
+
+			usb3@f0000 {
+				status = "okay";
+			};
+		};
+
+		nss_complex {
+			status = "okay";
+			pinctrl-0 = <&smi_nss_pins>;
+			pinctrl-names="default";
+		};
+
+		gop {
+			emac0: mac0 {
+				phy-mode = "rxaui";
+			};
+		};
+
+		pp3_platform {
+			status = "okay";
+
+			nic@0 {
+				status = "okay";
+				emac-data = <&emac0>;
+			};
+
+			nss@0 {
+				status = "okay";
+			};
+		};
+
+		pcie-controller {
+			status = "okay";
+			/*
+			 * The three PCIe units are accessible through
+			 * standard PCIe slots on the board.
+			 */
+			pcie@1,0 {
+				/* Port 0, Lane 0 */
+				status = "disabled";
+			};
+			pcie@2,0 {
+				/* Port 1, Lane 0 */
+				status = "okay";
+			};
+			pcie@3,0 {
+				/* Port 2, Lane 0 */
+				status = "disabled";
+			};
+			pcie@4,0 {
+				/* Port 3, Lane 0 */
+				status = "okay";
+			};
+		};
+	};
+};
-- 
1.9.1

